### PR TITLE
Improved README

### LINKED ISSUE #45: SRIOV feature not linked in to baremetal-deploy/README.md
Each feature should be linked in the top level README

### PR DESCRIPTION
**Pull request by e-minguez**
# Description

This PR revamps the README file to:
* explain the purpose of this repository
* add a section for installation artifacts
* add a section for the optional features with links to them
* add a how to contribute
* add a to-do

Fixes #45 #46 #47 #48 #49 #50 #51

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

- [x] Try to understand what the intention of this repo is before
- [x] Try to understand what the intention of this repo is after

**Test Configuration**:

- Versions: n/a
- Hardware: n/a

## Checklist

- [x] I have performed a self-review of my own code
- [x] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [x] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [x] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [x] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
-* Installation Troubleshooting
-* Installation scripts
-* Various feature yamls/manifest
+## Optional features
+
+* [Cluster Network Addons Operator](features/cnao/). This operator can be used to deploy additional
+networking components on top of Kubernetes/OpenShift cluster.
+* [CPU Manager](features/cpumanager/). CPU Manager manages groups of CPUs and constrains workloads
+to specific CPUs.
+* [Performance Tuning](features/performance). The manifests included in this folder enables
```
Comment (cynepco3hahue) [inline README.md]: via `Node Tuning` and `Machine config` operator

File: README.md [hunk 2]
```diff
@@ -1,17 +1,38 @@
...
+Huge Pages and Topology Manager via the 
+[Node Tuning Operator](https://github.com/openshift/cluster-node-tuning-operator)
+* [PTP](features/ptp). This operator manages cluster wide Precision Time Protocol (PTP) configuration.
+* [RT](features/rt). The manifests included in this folder install and enable a real time kernel as well
```
Comment (cynepco3hahue) [inline README.md]: we will merge this directory into performance

File: README.md (hunk 1 above)
Comment (e-minguez) [reply README.md]: Fixed, thanks

File: README.md (hunk 2 above)
Comment (e-minguez) [reply README.md]: Fixed, thanks

**Comment by e-minguez** [comment]
@cynepco3hahue I've added your suggestions. Do you mind to take a look?
Thanks!

**Comment by cynepco3hahue** [comment]
lgtm, but I do not have permissions to review on the repo root level :)

File: README.md [hunk 3]
```diff
@@ -1,17 +1,42 @@
...
+## Optional features
+
+* [CPU Manager](features/cpumanager/). CPU Manager manages groups of CPUs and constrains workloads
+to specific CPUs.
+* [PTP](features/ptp). This operator manages cluster wide Precision Time Protocol (PTP) configuration.
+* [SCTP](features/sctp). These assets enables Stream Control Transmission Protocol (SCTP) in the RHCOS
+worker nodes.
+* [SRIOV](features/sriov). The SR-IOV Network Operator creates and manages the components of the SR-IOV stack.
+
+## Performance uning
```
Comment (fromanirh) [inline README.md]: typo: should be "tuning"

Comment (e-minguez) [reply README.md]: Fixed!

File: README.md [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
+## Performance tuning
+
+The [Performance Tuning](features/performance) folder contains some assets intended to improve performance such as:
+
+* Huge Pages
+* Topology Manager
+* Real time kernel (including a new `worker-rt` Kubernetes/OpenShift node role)
+
+Those assets are applied by using the [Node Tuning](https://github.com/openshift/cluster-node-tuning-operator)
+and the [Machine Config](https://github.com/openshift-kni/baremetal-deploy/blob/master/features/sriov/deploy.sh) operators.
```
Comment (slintes) [inline README.md]: Machine Config operator should link to https://github.com/openshift/machine-config-operator.
The deploy script is a moving target at the moment, see #59

Comment (e-minguez) [reply README.md]: I've rephrased the explanation and fixed the link (bad copypasta!)

**Pull request merged**